commands/summary.go

```diff
@@ -274,6 +235,103 @@ func (cmd *summaryCommand) makeArchive() (err error) {
 	return //nolint:nakedret
 }
 
+type profilerPath struct {
+	suffix  string
+	webPath string
+}
+
+func (cmd *summaryCommand) getPprofData() []string {
+	profilerPaths := []profilerPath{
+		{
+			suffix:  "profile.pb.gz",
+			webPath: "/profile?seconds=60",
+		},
+		{
+			suffix:  "heap.pb.gz",
+			webPath: "/heap?gc=1",
+		},
+		{
+			suffix:  "trace.out",
+			webPath: "/trace?seconds=10",
+		},
+	}
+	apps := map[string]string{
+		"pmm-agent": "http://127.0.0.1:7777/debug/pprof",
+	}
+	if !cmd.SkipServer {
+		apps["pmm-managed"] = "http://127.0.0.1:7773/debug/pprof"
+		apps["qan-api2"] = "http://127.0.0.1:9933/debug/pprof"
+	}
+
+	out := make(chan string)
+	files := make([]string, 0)
+	wg := &sync.WaitGroup{}
+	wgr := &sync.WaitGroup{}
+
+	wgr.Add(1)
+
+	go func() {
+		for fileName := range out {
+			files = append(files, fileName)
+		}
+
+		wgr.Done()
+	}()
+
+	for appName, baseUrl := range apps {
+		wg.Add(1)
+		go func(appName, baseUrl string) {
+			defer wg.Done()
+
+			for _, file := range profilerPaths {
+				fs := fmt.Sprintf("%s-%s", appName, file.suffix)
+				url := baseUrl + file.webPath
+
+				downloadProfilerData(url, fs, out)
```

Comment:
We definitely should not fetch several profiles in parallel: it will give us skewed results in best, and virtually stop the process at worst, especially the simultaneously running CPU profiling and tracing.

Reply:
We actually don't fetch them in parallel for the same app. So we do, but we do for different apps.

Reply:
> Please add a comment before appName/baseURL loop to make it more noticeable.

That wasn't addressed.